include/common/BufferHelper.h

```diff
@@ -0,0 +1,16 @@
+#ifndef BUFFER_HELPER_H
+#define BUFFER_HELPER_H
+
+#include <cstdint>
+
+typedef struct AugmentedBuffer
```

Comment:
Why not just call this `Buffer`?

Reply:
Just because the convention is that buffers usually represent the pointer/array of memory. So having this struct called `AugmentedBuffer` is to avoid breaking that convention.